upgrade-1.x-to-2.x.sh

```diff
@@ -0,0 +1,410 @@
+	    ;;
+    esac
+fi
+
+# Check we are using the first root filesystem
+root_part=$(findmnt -n --raw --evaluate --output=source /)
+case $root_part in
+    *p2)
+	log "Current root partition is first root partition."
+	;;
+    *p3)
+	log "Current root partition $root_part is the second root partition. Copying existing OS to first partition..."
+	root_dev=${root_part%p3}
+	dd if=${root_dev}p3 of=${root_dev}p2
+	log "Updating bootloader to point to first partition..."
+	case $SLUG in
+	    beaglebone*)
+		sed -i -e 's/bootpart=1:3/bootpart=1:2/' /mnt/boot/uEnv.txt
+		;;
+	    raspberry*)
+		sed -i -e 's/mmcblk0p3/mmcblk0p2/' /mnt/boot/cmdline.txt
+		;;
+	esac
+	log "Rebooting..."
+	sync
+	reboot
+	;;
+    *)
+	log ERROR "Current root partition $root_part is not first or second root partition, aborting."
+	;;
+esac
+root_dev=${root_part%p2}
+
+# Check if we need to install some more extra tools
+if ! version_gt $VERSION $PREFERRED_HOSTOS_VERSION && ! [ "$VERSION" == $PREFERRED_HOSTOS_VERSION ]; then
+    log "Host OS version $VERSION is less than $PREFERRED_HOSTOS_VERSION, installing tools..."
+    tools_path=/home/root/upgrade_tools
+    mkdir -p $tools_path
+    export PATH=$tools_path:$PATH
+    case $BINARY_TYPE in
+	arm)
+	    download_uri=https://github.com/resin-os/resinhup/raw/onetotwo/upgrade-binaries/$BINARY_TYPE
+	    curl -s -L -o $tools_path/tar $download_uri/tar
+	    chmod 755 $tools_path/tar
+	    curl -s -L -o $tools_path/e2label $download_uri/e2label
+	    chmod 755 $tools_path/e2label
+	    curl -s -L -o $tools_path/mkfs.ext4 $download_uri/mkfs.ext4
+	    chmod 755 $tools_path/mkfs.ext4
+	    curl -s -L -o $tools_path/resize2fs $download_uri/resize2fs
+	    chmod 755 $tools_path/resize2fs
+	    ;;
+	*)
+	    log ERROR "Binary type $BINARY_TYPE not supported."
+	    ;;
+    esac
+fi
+
+# Stop docker containers, /var/lib/docker and /resin-data on p6
+log "Stopping all containers..."
+systemctl stop update-resin-supervisor.timer > /dev/null 2>&1
+systemctl stop resin-supervisor > /dev/null 2>&1
+$DOCKER stop $($DOCKER ps -a -q) > /dev/null 2>&1 || true
+log "Removing all containers..."
+$DOCKER rm $($DOCKER ps -a -q) > /dev/null 2>&1 || true
+systemctl stop docker
+
+# Switch connman to use bind mounted state dir
+log "Switching connman to bind mounted state dir..."
+mkdir -p /tmp/connman
+cp -a /var/lib/connman/* /tmp/connman/
+# Versions before 1.20 need this to prevent dropping VPN
+sed -i -e 's/NetworkInterfaceBlacklist=docker,veth,tun,p2p/NetworkInterfaceBlacklist=docker,veth,tun,p2p,resin-vpn/' /etc/connman/main.conf
+mount -o bind /tmp/connman /var/lib/connman
+systemctl restart connman
+
+# Save /resin-data to rootB
+mkfs.ext4 -F ${root_dev}p3
+mkdir -p /tmp/backup
+mount ${root_dev}p3 /tmp/backup
+(cd /mnt/data; tar -zcf /tmp/backup/resin-data.tar.gz resin-data)
+
+# Unmount p6
+log "Unmounting filesystems..."
+umount /mnt/data
+umount /var/lib/docker
+umount /resin-data
+
+# Save conf contents to /boot
+# XXX
```

Comment:
It means not implemented. I though this was something that might need doing but I don't really have any further information on it.